service/matching/version_rule_helpers.go

```diff
@@ -489,12 +486,17 @@ func given2ActualIdx(idx int32, rules []*persistencespb.AssignmentRule) int {
 	return -1
 }
 
-// validRamp returns true if the percentage ramp is within [0, 100), or if the ramp is nil
-func validRamp(ramp *taskqueue.RampByPercentage) bool {
+// isValidRamp returns true if the percentage ramp is within [0, 100], or if the ramp is nil
+func isValidRamp(ramp *taskqueue.RampByPercentage) bool {
```

Comment:
I think it helps if we convert nil ramps to 100% ramps when we persist them (or when we return the rules in GetRules response). That way user wouldn't be confused if nil means 100% or 0%.